xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-macro/xwiki-platform-notifications-macro-ui/src/main/resources/XWiki/Notifications/Code/Macro/NotificationsMacro.xml

```diff
@@ -423,9 +423,9 @@ require(['jquery', 'XWikiAsyncNotificationsMacro', 'xwiki-events-bridge'], funct
         // Style the read button
         readButton.addClass('notification-event-read-button').addClass('btn btn-xs');
         // Insert the cross icon and an accessible label
-        var readButtonIcon = $('$services.icon.renderHTML("check")');
+        var readButtonIcon = $("$escapetool.javascript($services.icon.renderHTML('check'))");
```

Comment:
I see you've modified as suggested by @surli . Does it mean it wasn't working before your change (since you just reused what was there)?

Reply:
The previous code was working at the cost of an extra string evaluation which was not needed indeed, as pointed out by @surli. This commit fixes this, adding also a JavaScript escape which is needed for escaping quotes possibly present (typically there are some quotes in the HTML icon code, hence a plain `$("$services.icon.renderHTML('check')")` would not work).